### ISSUE TITLE
Poudriere does not clean up datasets if an error occured during jail creation

### ISSUE THREAD

**Issue by 0mp** [freebsd/poudriere#855]
## Prerequisites
- [x] Have you checked for an existing issue describing your problem?
- [x] Are you running the latest version?
- [x] Is your ports tree recent?
- [x] Is your FreeBSD Host on a [supported release](https://www.freebsd.org/security/#sup)?

## Describe the bug

Poudriere does not clean up properly if there is an error during a jail creation.

## How to reproduce
1. Try to create a new jail with `sudo poudriere jail -c -j 130a -v 13.0-RELEASE -a amd64`
2. See if it fails with
    > Error: Failed to fetch from https://download.FreeBSD.org/ftp/releases/amd64/amd64/13.0-RELEASE/MANIFEST
3. If so, then try to create the jail again.
4. You won't be able to do it, as the ZFS dataset already exists.

## Expected behavior

If an error occurs during the jail creation, everything that's been created in the process should be cleaned up.

## Screenshots

Here's the command log:

```console
$ sudo poudriere jail -c -j 130a -v 13.0-RELEASE -a amd64
[00:00:00] Creating 130a fs at /usr/local/poudriere/jails/130a... done
mkdir: /usr/local/poudriere/jails/130a: No such file or directory
[00:00:00] FREEBSD_HOST from config invalid; defaulting to https://download.FreeBSD.org
[00:00:00] Fetching MANIFEST for FreeBSD 13.0-RELEASE amd64
fetch: /usr/local/poudriere/jails/130a/fromftp/MANIFEST: open(): No such file or directory
fetch: /usr/local/poudriere/jails/130a/fromftp/MANIFEST: open(): No such file or directory
[00:00:01] Error: Failed to fetch from https://download.FreeBSD.org/ftp/releases/amd64/amd64/13.0-RELEASE/MANIFEST
[00:00:01] Error while creating jail, cleaning up.
[00:00:01] Removing 130a jail... done
[00:00:01] Cleaning 130a data... done
$ sudo poudriere jail -c -j 130a -v 13.0-RELEASE -a amd64
[00:00:00] Creating 130a fs at /usr/local/poudriere/jails/130a...cannot create 'zroot/poudriere/jails/130a': dataset already exists
 fail
[00:00:00] Error: Failed to create FS zroot/poudriere/jails/130a
$ sudo poudriere jail -d -j 130a
[00:00:00] Error: No such jail: 130a
```

## Environment
 - Host OS [e.g. 12.2 amd64]: 14.0-CURRENT amd64
 - Jail OS [e.g. 12.0 powerpc]: 13.0-RELEASE amd64 
 - Poudriere Version [e.g. 3.3.1 or git hash or port version]: poudriere-devel-3.3.99.20210303_1
 - Ports branch and revision [e.g. 2020Q3 r550754]: latest

## Additional context
<Add any other context about the problem here.>

**Comment by bdrewery** [comment]
I think 1c9357d08e35f2bb248533ce043b2e61920ebf9e fixed this. Can you test please? If that helps I'll push it out to the port ASAP as it keeps coming up.

**Comment by 0mp** [comment]
I've rebuilt poudriere-devel-3.3.99.20210303_1 with the patch applied and removed the leftover dataset with `sudo zfs destroy zroot/poudriere/jails/130a`. Then I was able to create a poudriere jail. The patch seems to be working.

**Comment by brd** [comment]
Works for me too.

**Comment by bdrewery** [comment]
Fixed in freebsd/freebsd-ports@b42828e7fd3e poudriere-devel-3.3.99.20210303_2
